packages/trace-viewer/src/sw.ts

```diff
@@ -145,6 +146,49 @@ async function doFetch(event: FetchEvent): Promise<Response> {
   return snapshotServer.serveResource(lookupUrls, request.method, snapshotUrl);
 }
 
+function headersForResource(traceModel: TraceModel, sha1: string): Headers | undefined {
+  const attachment = traceModel.attachmentForSha1(sha1);
+  if (!attachment)
+    return;
+  const headers = new Headers();
+  let name = attachment.name;
+  const nameHasExt = name.includes('.');
+  if (!nameHasExt)
+    name += extensionForContentType(attachment.contentType);
+  headers.set('Content-Disposition', `attachment; filename="${name}"`);
+  if (attachment.contentType)
+    headers.set('Content-Type', attachment.contentType);
+  return headers;
+}
+
+function extensionForContentType(contentType: string): string {
```

Comment:
Don't we have an utility like this already?

Reply:
did not find any.

Reply:
mime.getExtension() should do what you need